site/en/education/recommendations-for-web-views.md

```diff
@@ -0,0 +1,48 @@
+
+## General App Development Advice
+
+### Reduce or remove use of web browser content
+
+By reducing the amount of browser based content shown in your application, you can reduce the likelihood of unwanted content being accessible within your application. If this is not possible, then ensure your WebView only serves content you want and does not link out to a search engine or the open web.
+
+### Ensure your WebView only serves content you want and does not link out
+
+If your app requires that a webview is within your application, make sure that the content within your webview is not able to escape your application and navigate to content that you do not own or control. This can be done by modifying the [webview client to override loading certain URLs via an allowlist](<https://developer.android.com/reference/android/webkit/WebViewClient#shouldOverrideUrlLoading(android.webkit.WebView,%20android.webkit.WebResourceRequest)>) on Android or by manually parsing through your site and validating that the links do not link out to third party content that you cannot control as this may enable your users to go to inappropriate content. For Chrome Apps, you can use the [WebRequestEventInterface](https://developer.chrome.com/docs/extensions/reference/webviewTag/#type-WebRequestEventInterface) to also modify requests and validate that they are allowed to navigate to your allowed list of sites.
+
+## Android Apps
+
+### Use Chrome Custom Tabs instead of WebViews
+
+By employing Chrome Custom Tabs, you are simply redirecting your users to the Chrome Browser on the device that they are using. This is most useful for when you may have a third party identity provider that relies on implementing OAuth in a webview. For Chrome OS, this will launch the normal browser and allow users to navigate within the page as normal but it will allow the enforcement of browser policies to the content that they students are using. For most educational organizations, this means that the content that would be displayed would be visible to extensions running on the browser including content filtering.
+
+### Using Google Sign In without an additional identity provider
+
+If your Android application only uses Google Sign in, do not redirect to the webview to sign in but rather handle it with the [One Tap sign-in and sign-up flow](https://developers.google.com/identity/one-tap/android). Links in this sign-in flow start a browser intent rather than an embedded webview so your data can be viewed in the Chrome browser of the device, not a webview which may not have the same filtering controls applied to it.
```

Comment:
```suggestion
If your Android application only uses Google Sign in, do not redirect to the WebView to sign in but rather handle it with the [One Tap sign-in and sign-up flow](https://developers.google.com/identity/one-tap/android). Links in this sign-in flow start a browser intent rather than an embedded WebView so your data can be viewed in the Chrome browser of the device, not a WebView which may not have the same filtering controls applied to it.
```